db/sqlalchemy.py

```diff
@@ -15,8 +17,11 @@ def __init__(
 
     def make_declarative_base(self, model, metadata=None):
         """Creates or extends the declarative base."""
-        if self.Model is None:
-            self.Model = super().make_declarative_base(model, metadata)
+        try:
+            if self.Model is None:
+                self.Model = super().make_declarative_base(model, metadata)
 
-        set_query_property(self.Model, self.session)
-        return self.Model
+            set_query_property(self.Model, self.session)
+            return self.Model
+        except Exception as e:
+            logger.exception("Error during declarative base creation: %s", e)
```

Comment:
This would result in the exception not bubbling up to a higher level in the stack. It's always preferable for exceptions to be thrown up as high as possible.
Additionally, handling the exception at this level does not help. The exception must bubble to a higher level in the stack where it can be handled or logged appropriately.